source/macros.tex

```diff
@@ -214,6 +214,11 @@
 \newcommand{\libmember}[2]{\indexlibrarymember{#1}{#2}#1}
 \newcommand{\libspec}[2]{\indexlibrarymemberx{#1}{#2}#1}
 
+% index for macros
+% use in-place defining the macro in header synopses
+% initially just a global entry in the library index
+\newcommand{\libmacro}[1]{\indexlibraryglobal{#1}#1}
```

Comment:
So, please put this commit plus the definition of \libmacrox (which I can't find right now) into a separate starter commit with label [macros].  Then, we mostly have nice follow-up commits for individual header files, and those commits can carry meaningful labels.

Oh, and \libmacrox is a bit of a misnomer, because "...x" for the indexing macros usually means you can select the stuff that goes into the index separately from the stuff that is shown where the macro appears in LaTeX.  But your \libmacrox is not like that; it prepends double underscores (I guess?).  So, please use a different suffix. Maybe \libmacrous ("us" for underscore) or whatever.

There is also the option to define a local LaTeX macro just for the `<version>` #defines , which also includes the "cpp_lib" prefix and saves redundancy there. But maybe that's a bit too much for this round of edits.